File: backend/ProcesosApi/Models/Proceso.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ProcesosApi.Models;

public class Proceso
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = null!;
    public DateTime Fecha { get; set; }
    public string Demandante { get; set; } = string.Empty;
    public string Demandado { get; set; } = string.Empty;
    public string Radicado { get; set; } = string.Empty;
    public string Juzgado { get; set; } = string.Empty;
    public string Ciudad { get; set; } = string.Empty;
    public string ClaseProceso { get; set; } = string.Empty;
    public string? Representamos { get; set; }
    public string? ProcesoIngresadoPor { get; set; }
    public string? Honorarios { get; set; }
    public List<Comision> Comisiones { get; set; } = new();
    public string? EstadoActual { get; set; }
    public Dictionary<string, string> CamposGlobales { get; set; } = new();
    public List<CampoPropio> CamposPropios { get; set; } = new();
    public bool Terminado { get; set; }
    public string? NotaTerminacion { get; set; }
    public DateTime? FechaTerminacion { get; set; }
    public DateTime FechaCreacion { get; set; }
    public DateTime FechaActualizacion { get; set; }
    public bool Activo { get; set; } = true;

    public List<Pago> Pagos { get; set; } = new();
    public List<Nota> Notas { get; set; } = new();
    public List<Recordatorio> Recordatorios { get; set; } = new();
}

File: frontend/src/types/proceso.ts
import type { Pago } from './pago';
import type { Nota } from './nota';
import type { Recordatorio } from './recordatorio';

export interface Comision {
  persona: string;
  porcentaje: string;
}

export interface CampoPropio {
  nombre: string;
  valor: string;
}

export interface ProcesoListItem {
  id: string;
  fecha: string;
  demandante: string;
  demandado: string;
  radicado: string;
  ciudad: string;
  claseProceso: string;
  estadoActual: string | null;
  procesoIngresadoPor: string | null;
  terminado: boolean;
  cantidadPagos: number;
  cantidadNotas: number;
  recordatoriosPendientes: number;
}

export interface ProcesoDetalle {
  id: string;
  fecha: string;
  demandante: string;
  demandado: string;
  radicado: string;
  juzgado: string;
  ciudad: string;
  claseProceso: string;
  representamos: string | null;
  procesoIngresadoPor: string | null;
  honorarios: string | null;
  comisiones: Comision[];
  estadoActual: string | null;
  camposGlobales: Record<string, string>;
  camposPropios: CampoPropio[];
  terminado: boolean;
  notaTerminacion: string | null;
  fechaTerminacion: string | null;
  fechaCreacion: string;
  fechaActualizacion: string;
  pagos: Pago[];
  notas: Nota[];
  recordatorios: Recordatorio[];
}

export interface ProcesoCreate {
  fecha: string;
  demandante: string;
  demandado: string;
  radicado: string;
  juzgado: string;
  ciudad: string;
  claseProceso: string;
  representamos?: string;
  procesoIngresadoPor?: string;
  honorarios?: string;
  comisiones: Comision[];
  estadoActual?: string;
  camposGlobales?: Record<string, string>;
  camposPropios?: CampoPropio[];
  terminado?: boolean;
  notaTerminacion?: string;
}

export interface PaginatedResult<T> {
  items: T[];
  totalCount: number;
  pagina: number;
  porPagina: number;
  totalPaginas: number;
}

export interface ProcesoFiltros {
  busqueda?: string;
  ciudad?: string;
  claseProceso?: string;
  estado?: string;
  ingresadoPor?: string;
  ordenarPor?: string;
  ordenDescendente?: boolean;
  pagina?: number;
  porPagina?: number;
}
